api/ondemand_vhds_rds_protocol.rst

```diff
@@ -0,0 +1,187 @@
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with their route config names + domains provided in the
+resource_names_unsubscribe field. Envoy will remove any route config names +
+domains that it finds in the
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+removed_resources field.
+
+Subscribing to Virtual Hosts
+============================
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the resource_names_subscribe field populated with the route config names
++ domains of each of the resources that it would like to subscribe to. Each of
+the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+During spontaneous updates configuration server will only send updates for
+virtual hosts that Envoy is aware of -- the configuration server needs to
+keep track of virtual hosts known to Envoy.
+
+Updates of non-virtual host fields
+==================================
+It might be required to update non-virtual host fields of route configuration
+while keeping virtual hosts as is. This would probably happen during
+spontaneous updates only. In such a case a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+containing a route configuration with an empty virtual hosts list can be used.
+
+Compatibility with Scoped RDS
+=============================
+Both approaches appear to be compatible with scoped RDS proposal:
+route_configuration name can still be used for vhost matching, but with scoped
+RDS configured it would point to a scoped route configuration.
+
+The xDS management server will also support the ability of Envoy to tell it
+when a resource hasn't been used and is safe to stop monitoring. The resources
+that can be removed include the base resources that xDS management server
+initially sent Envoy.
+
+Any future updates to this resource will be detected by xDS management server
+as part of the normal xDS flow and sent along with normal cluster updates.
+
+Full flow example
+==================
+
+.. figure:: diagrams/delta_rds_full_flow.svg
+   :alt: Delta RDS full flow
+
+Example xDS management server-generated RDS
+===========================================
+.. code:: yaml
+
+  '2001':
+    name: '2001'
+    virtual_hosts:
+    - name: s1http.none:2001
+      domains:
+      - s1http.none
+      - s1http.none:2001
+      routes:
+      - match:
+          PathSpecifier:
+            Prefix: "/"
+        Action:
+          Route:
+            ClusterSpecifier:
+              Cluster: outbound|2001||s1http.none
+            HostRewriteSpecifier:
```

Comment:
This is a weird mix of camel case and underscore separated fields.

Reply:
Removed the example since it didn't add much value.